### ISSUE TITLE
Add possibility to query HistoricVariableInstances

### ISSUE THREAD

**Issue by berndruecker** [camunda/camunda-bpm-assert#42]
If a process instance is ended it should be possible to query at least the latest value of a process variable.
I think it would be OK to do that internally in the "variables()" assert - just return the map of HistoricVariableInstances if a ProcessInstance is ended. Or create a new "historicVariables()"?

**Comment by kifj** [comment]
Hi Bernd

I had the same problem, and my approach is the historicVariables() method, even without reading your post; see pull request #43.

Joe

**Comment by martinschimak** [comment]
Hi all, I reviewed this in more detail now. Sorry for the delay! Even though I agree that a historicVariables() method looks like a more robust/safe/obvious/less-magic approach at first, I went now for the approach to include that functionality within the existing variables() method. 

I added the functionality not just for the variables() map, but also for the two other variables related assertions hasVariables(String... keys) as well as hasNoVariables(). I also added many testcases for all those "historic" cases and a warning that users must use "audit" level at least in case they want to use those assertions for historic instances.

The main reasoning for going for the "hidden" approach is that I do not see any particular harm in adding this to existing methods, but do see a certain upside when writing "chunked" tests for heavily changing process definitions: the inspection of variables works the same way at a particular "wait state", no matter whether it is a real waitstate of a running instance or a final "waitstate" because the instance already finished at that point. As a consequence, when changing process definitions, I would expect less unit tests breaking for the "wrong reasons": when checking for the state of variables, I am not interested at which particular point I am at the moment, but just in the state of the variables. So, if that state is ok, the assertion should pass. If I want to explicitely check for the position in the running/historic process instance, I can/should do that by using other assertions.
